core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -929,10 +848,7 @@ public ReplicableCommand copyForRemote(ReadOnlyManyCommand command, List<Object>
 
       @Override
       public void applyLocalResult(MergingCompletableFuture allFuture, Object rv) {
-         Supplier<ArrayIterator> supplier = () -> new ArrayIterator(allFuture.results);
-         BiConsumer<ArrayIterator, Object> consumer = ArrayIterator::add;
-         BiConsumer<ArrayIterator, ArrayIterator> combiner = ArrayIterator::combine;
-         ((Stream) rv).collect(supplier, consumer, combiner);
+         ((Stream) rv).collect(new ArrayCollector(allFuture.results));
```

Comment:
Yes, ReadOnlyManyCommand.perform() returns a Stream. If we get something else, there's not much else to do about it than throw an exception.